### DISCUSSION TITLE
Docker Images (FROM ...) are not getting updated properly

### DISCUSSION THREAD

**Discussion by DManstrator** [renovatebot/renovate#18503] [Request Help]
Hello.

We currently use [`eclipse-temurin:17.0.4_8-jre-focal`](https://registry.hub.docker.com/layers/library/eclipse-temurin/17.0.4_8-jre-focal/images/sha256-f2f91c0d1f23c6e75b6f66e8a9db62dd4b1c100598fed75e6742a8a0b20392e0) for our Docker Images (`FROM eclipse-temurin:17.0.4_8-jre-focal`).

Recently, [`eclipse-temurin:17.0.4.1_1-jre-focal`](https://registry.hub.docker.com/layers/library/eclipse-temurin/17.0.4.1_1-jre-focal/images/sha256-55c1ff40a91898c456dc5a874a66db17b6aa1b4aef8d9cac1ad3b194575638a1) was released but Renovate didn't create a MR for it. The reason for that is that is most likely the RegEx we defined.

Currently, we have the following setting in out `renovate.json`:
```json
{
  "matchPackageNames": [
    "eclipse-temurin"
  ],
  "versioning": "regex:^(?<major>\\d+)(\\.(?<minor>\\d+))?(\\.(?<patch>\\d+))?(_(?<build>\\d+))?(-(?<compatibility>.*))?$"
}
```

The RegEx doesn't contain the fourth version digit (after patch) so the RegEx doesn't match. Instead it should be like that:
`^(?<major>\d+)(\.(?<minor>\d+))?(\.(?<patch>\d+))?(\.(?<whatToPassHere>\d+))?(_(?<build>\d+))?(-(?<compatibility>.*))?$`

My problem is that I don't know how to name the fourth digit (`whatToPassHere`) so Renovate knows what to compare to. I also didn't find a fitting name in the docs.

Can you tell me how I would need to adjust the RegEx to include such versions?

Kind regards,

Daniel

Edit: Here some logs in case they help:

```log
 "dockerfile": [
   {
     "packageFile": "Dockerfile",
     "deps": [
       {
         "depName": "eclipse-temurin",
         "currentValue": "17.0.4_8-jre-focal",
         "replaceString": "eclipse-temurin:17.0.4_8-jre-focal",
         "autoReplaceStringTemplate": "{{depName}}{{#if newValue}}:{{newValue}}{{/if}}{{#if newDigest}}@{{newDigest}}{{/if}}",
         "datasource": "docker",
         "depType": "final",
         "depIndex": 0,
         "updates": [
           {
             "bucket": "major",
             "newVersion": "19_36-jre-focal",
             "newValue": "19_36-jre-focal",
             "newMajor": 19,
             "newMinor": 0,
             "updateType": "major",
             "branchName": "renovate/eclipse-temurin-19.x"
           }
         ],
         "warnings": [],
         "versioning": "regex:^(?<major>\\d+)(\\.(?<minor>\\d+))?(\\.(?<patch>\\d+))?(_(?<build>\\d+))?(-(?<compatibility>.*))?$",
         "currentVersion": "17.0.4_8-jre-focal",
         "isSingleVersion": true,
         "fixedVersion": "17.0.4_8-jre-focal"
       }
     ]
   }
 ],
```

It seems that `17.0.4.1_1-jre-focal` wouldn't be used even the RegEx would match since `newValue` is already set to 19. We didn't updated to that on purpose. Perhaps we could pin it to major `17` so it could be detected again?

From what I read in the [docs](https://docs.renovatebot.com/docker/#version-upgrading), "Renovate will upgrade minor/patch versions (like from 1.2.0 to 1.2.1), but not upgrade major versions. If you wish to enable major versions then add the preset docker:enableMajor to your extends array in your renovate.json". However, we don't have that preset enable. Perhaps this happened due to how we treat it.

**Comment by Gabriel-Ladzaretti** [comment]
Try using [loose versioning](https://docs.renovatebot.com/modules/versioning/#loose-versioning) in your regex manager.

**Reply by Gabriel-Ladzaretti** [reply]
here's an example 
https://github.com/renovatebot/renovate/discussions/18137#discussioncomment-3864065

**Reply by DManstrator** [reply]
Hey Gabriel,

thanks for the input. I looked up the example and the Docs and I don't think this is going to work. Also found the "docker" versioning while being in the docs and this shouldn't work either.

Our Dockerfile uses / gets the image with `FROM eclipse-temurin:17.0.4_8-jre-focal`. In the example, the version to update is a variable and gets restricted with a comment `# renovate: datasource=...` above. So I don't think this can be applied here.

Still thanks for your idea. :D
